### ISSUE TITLE
can git-sync suppport LFS?

### ISSUE THREAD

**Issue by willzhang** [kubernetes/git-sync#174]

**Comment by stp-ip** [comment]
Currenly it does not support LFS.

**Comment by willzhang** [comment]
close

**Comment by rjanovski** [comment]
any chance lfs can be added?

**Comment by thockin** [comment]
I'm not against it in principle, but I have never used it and don't know
much about it.  Can you write more about how you think it would work?

**Comment by thockin** [comment]
I know nothing about it - is it as simple as installing git-lfs in the
image?  Help me - write a testcase in test_e2e.sh ?

**Comment by rjanovski** [comment]
well,  pretty much.
- installing git-lfs (debian):
   - curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | sudo bash
   - sudo apt-get install git-lfs
   - git lfs install
- git lfs pull (in a git cloned folder, brings the large files as well)

**Comment by thockin** [comment]
Does the LFS pull use the current repo hash or is that somehow
independent?  How does that work and how does it interact with things like
--depth?

The most useful thing you could do would be to show me an end-to-end
example of setting up a repo with LFS files, cloning that repo, then
pulling lfs, them updating the repo and refreshing it in the clone.

**Comment by bmfisher** [comment]
@rjanovski @thockin 
I was able to extend the image to install git-lfs and successfully pull / sync lfs files by mounting the appropriate config.  Out setup is similar to the [proxy](https://github.com/kubernetes/git-sync/blob/release-3.x/docs/proxy.md) setup described in the docs.  To enable LFS, I added the files `/tmp/.gitconfig` and `/tmp/.git-credentials` to inject credentials for our self hosted git-lfs repository.  I injected these files with Kubernetes secrets.  My git-sync pod is up and running and the lfs files appear to have been completely pulled.  I also specified the `http_proxy` environment variable as the git-lfs pull wouldn't know about the one I specified in the ssh config.

Hope this helps!

.gitconfig:
```
[credential]
    helper = store
```

.git-credentials:
```
https://user:password@our.domain/repository/git-lfs
```

Dockerfile:
```
FROM k8s.gcr.io/git-sync/git-sync:v3.1.7
USER root
RUN apt-get update
RUN apt-get -y install git-lfs
USER 65533:65533
```
